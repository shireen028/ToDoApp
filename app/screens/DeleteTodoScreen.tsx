import React, {useState} from 'react';
import {View, StyleSheet, Alert,Image} from 'react-native';
import {Text, Button} from 'react-native-elements';
import {Todo} from '../types';
import {Navigation} from 'react-native-navigation';

export type Props = {
  todo: Todo;
  componentId: string;
  removeTodo: Function;
};
export const DeleteTodoScreen: React.FC<Props> = ({
  todo,
  componentId,
  removeTodo,
}) => {
  const [loading, setLoading] = useState(false);
  const deleteTodo = async () => {
    setLoading(true);
    try {
      const response = await fetch(
        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
        {
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            'Cache-Control': 'no-cache',
          },
          method: 'DELETE',
        },
      );
      await response.json();
      removeTodo();
    } catch (error) {
      Alert.alert('Error', 'Delete todo failed');
    }

    setLoading(false);
    Navigation.pop(componentId);
  };
  return (
    <View style={styles.container}>
      <Text h3 style={styles.title}>
        Delete Todo
      </Text>
      <View style={styles.content}>
  
    <Image style={{width: 40, height: 40 }} source= 
         {todo.completed? require('assets/done.png') : 
         require('assets/inprogress.png') }/>

        <Text h4 style={styles.title}>
          {todo.title}
        </Text>
      </View>
      <Button
        title="Delete"
        loading={loading}
        onPress={deleteTodo}
        icon={styles.icon}
        buttonStyle={styles.button}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 15,
  },
  title: {
    textAlign: 'center',
    marginVertical: 20,
  },
  content: {
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    height: '60%',
    borderRadius: 10,
  },
  button: {
    minHeight: 52,
    minWidth: 350,
    backgroundColor: 'red',
    marginVertical: 50,
  },
  icon: {
    name: 'trash-outline',
    type: 'ionicon',
    color: 'white',
  },
});
