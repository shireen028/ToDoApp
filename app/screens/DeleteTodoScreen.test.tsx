import 'react-native';
import React from 'react';
import {DeleteTodoScreen} from './DeleteTodoScreen';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const todo = {
    id: 1,
    title: 'awesome',
    userId: 2,
    completed: true,
  };
  const tree = renderer.create(
    <DeleteTodoScreen componentId="sdfs" todo={todo} removeTodo={() => {}} />,
  );
  expect(tree).toMatchSnapshot();
});
