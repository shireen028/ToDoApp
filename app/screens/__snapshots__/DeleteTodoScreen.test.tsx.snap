// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
        "fontSize": 36,
        "marginVertical": 20,
        "textAlign": "center",
      }
    }
  >
    Delete Todo
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "60%",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/done.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 28.285714285714285,
          "marginVertical": 20,
          "textAlign": "center",
        }
      }
    >
      awesome
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 50,
            "minHeight": 52,
            "minWidth": 350,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 5,
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                Ôñç
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
